lib/puppet/parser/functions/fqdn_rand.rb

```diff
@@ -17,7 +20,12 @@
   node. (For example, `fqdn_rand(30)`, `fqdn_rand(30, 'expensive job 1')`, and
   `fqdn_rand(30, 'expensive job 2')` will produce totally different numbers.)") do |args|
     max = args.shift.to_i
- 
+    initial_seed = args.shift
+    downcase = !!args.shift
+
+    fqdn = self['::fqdn']
+    fqdn.downcase! if downcase
```

Comment:
This is raising `Error: Evaluation Error: Error while evaluating a Function Call, can't modify frozen String` for me, trying to run in locally as
```
bx puppet apply -e 'notice(fqdn_rand(100, "expensive job 1", true))'
```

Reply:
Nice catch! Updated